src/store/diagnostics.ts

```diff
@@ -14,6 +14,23 @@ interface DiagnosticsState {
 
 const uri = (port: number, pathRemainder: string) => `http://localhost:${ port }/v0/${ pathRemainder }`;
 
+/**
+ * Updates the muted property for diagnostic results.
+ * @param checks A collection of diagnostic results that require muting.
+ * @param mutedChecks A collection of key, value pairs that contains a key of
+ * the ID for the diagnostic and a boolean value for muting the result.
+ * @returns A collection of diagnostic results with an updated muted property.
+ */
+const mapMutedDiagnostics = (checks: DiagnosticsResult[], mutedChecks: Record<string, boolean>) => {
+  return checks.map((check) => {
+    if (Object.keys(mutedChecks).includes(check.id)) {
+      check.mute = mutedChecks[check.id];
```

Comment:
We're modifying this in-place; is that okay?

Reply:
good catch, this is not okay..